modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/TableCell.js

```diff
@@ -75,48 +75,48 @@ function TableCell({
 	valuePath,
 	view,
 }) {
-	let dataRenderer = DefaultRenderer;
-
 	const {customDataRenderers, inlineEditingSettings} = useContext(
 		DataSetContext
 	);
 
+	const [loading, setLoading] = useState(false);
+
 	const contentRenderer = view.contentRenderer;
 
+	let SyncDataRenderer = DefaultRenderer;
+
 	if (contentRenderer) {
 		if (customDataRenderers && customDataRenderers[contentRenderer]) {
-			dataRenderer = customDataRenderers[contentRenderer];
+			SyncDataRenderer = customDataRenderers[contentRenderer];
 		}
 		else {
-			dataRenderer = getDataRendererById(view.contentRenderer);
+			SyncDataRenderer = getDataRendererById(contentRenderer);
 		}
 	}
 
 	if (view.contentRendererModuleURL) {
-		dataRenderer = null;
+		SyncDataRenderer = null;
 	}
 
-	const [currentView, setCurrentView] = useState({
-		...view,
-		Component: dataRenderer,
-	});
-
-	const [loading, setLoading] = useState(false);
+	const [DataRenderer, setDataRenderer] = useState(() => SyncDataRenderer);
 
 	useEffect(() => {
-		if (!loading && currentView.contentRendererModuleURL) {
+		if (!loading && view.contentRendererModuleURL && !DataRenderer) {
 			setLoading(true);
-			getDataRendererByURL(currentView.contentRendererModuleURL).then(
-				(Component) => {
-					setCurrentView({
-						...currentView,
-						Component,
-					});
+
+			getDataRendererByURL(view.contentRendererModuleURL)
+				.then((Component) => {
+					setDataRenderer(() => Component);
```

Comment:
Can we simplify to `setDataRenderer(Component);`? The same with `setDataRenderer(() => null);` below.

Reply:
Unfortunately no, passing a functional component means passing a function that react will read as a setter function.

this is a common scenario
```
setDataRenderer((prevState) => {
   return changeSomething(prevState)
})
```

But it is what will happen
```
setDataRenderer((props) => {
   return (<div>{props.id}</div>)
})
```

Leading `DataRenderer` not to contain a component, but what the component returns